src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseTestCaseService.java

```diff
@@ -91,9 +92,10 @@ public ProgrammingExerciseTestCaseService(ProgrammingExerciseTestCaseRepository
             updatedTests.add(matchingTestCase);
         }
 
-        // Make sure that at least one test has a weight so that students can still achieve 100% score
+        // Make sure that at least one test has a weight >0 for purely automatic feedback so that students can still achieve 100% score.
+        // If manual feedback is given, then a test case weight of zero is okay, as students can still receive points via manual feedbacks.
         var testWeightsSum = existingTestCases.stream().mapToDouble(testCase -> Optional.ofNullable(testCase.getWeight()).orElse(0.0)).sum();
-        if (testWeightsSum <= 0) {
+        if (testWeightsSum < 0 || (testWeightsSum == 0 && programmingExercise.getAssessmentType() == AssessmentType.AUTOMATIC)) {
```

Comment:
I assume this should also fail if the Assessment is automatic but allows complaints? Would that be semi automatic then? Or is this independent of this?

Reply:
Automatic and complaints is independent from semi automatic. In `programming-exercise-lifecycle.component.ts` I found this explanation that seems to align with implementations e.g. in `ProgrammingAssessmentService`: SEMI_AUTOMATIC After all automatic tests have been run, the tutors will have to make a final manual assessment.